flocker/control/test/test_model_change.py

```diff
@@ -0,0 +1,418 @@
+        """
+        If the type of the key of a ``PMap`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPMap):
+            __key_type__ = Subtype
+            __value_type__ = int
+
+        class Different(CheckedPMap):
+            __key_type__ = ChangedSubtype
+            __value_type__ = int
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pmap_value_type_changed(self):
+        """
+        If the type of the value of a ``PMap`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPMap):
+            __key_type__ = int
+            __value_type__ = Subtype
+
+        class Different(CheckedPMap):
+            __key_type__ = int
+            __value_type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pset_value_new_type(self):
+        """
+        If the type of the value of a ``PSet`` changes the output changes.
+        """
+        class Original(CheckedPSet):
+            __type__ = int
+
+        class Different(CheckedPSet):
+            __type__ = str
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pset_value_type_changed(self):
+        """
+        If the type of the value of a ``PSet`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPSet):
+            __type__ = Subtype
+
+        class Different(CheckedPSet):
+            __type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pvector_value_new_type(self):
+        """
+        If the type of the value of a ``PVector`` changes the output changes.
+        """
+        class Original(CheckedPVector):
+            __type__ = int
+
+        class Different(CheckedPVector):
+            __type__ = str
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pvector_value_type_changed(self):
+        """
+        If the type of the value of a ``PVector`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPVector):
+            __type__ = Subtype
+
+        class Different(CheckedPVector):
+            __type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+
+def persist_model():
+    """
+    Store the current model to disk.
+
+    We also store the git hash of current checkout, so it's clear what
+    version of code was used to generate the model.
+    """
+    git_hash = check_output(
+        [b"git", b"show", b'--format=%H', b"-s"]).strip()
```

Comment:
If there are changes to the working copy, they'll change the output of this function but they won't be reflected in this git hash.

Perhaps `flocker.__version__`?  I think it's strictly more informational.